RCT/JX/Rule21-8

Jiarongx-Xie opened the pull request:
This rule has been tested.

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+    def is_applicable(self, context, data=None):
+        rmi_b = context.BASELINE_0
+        hw_loop_zone_list_w_area_dict_b = get_hw_loop_zone_list_w_area(rmi_b)
+        boiler_loop_ids_b = find_all("boilers[*].loop", rmi_b)
```
Could we add ```$``` in the json path as shown below? 

```python
boiler_loop_ids = find_all("$.boilers[*].loop", rmi_b)
```

This works without ```$``` because there is a function that adds ```$```. However, we'll likely remove the function and it's recommended to have a full path (including $).

Jiarongx-Xie replied:
Revised! Thank you for the clarification:)

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+    def create_data(self, context, data):
+        rmi_b = context.BASELINE_0
+        boiler_loop_ids = find_all("boilers[*].loop", rmi_b)
+        return {"loop_boiler_dict": boiler_loop_ids}
```
I believe ```loop_boiler_dict``` is a list. Could you double-check this? Also, you can add ```_b``` in the variable name.

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+
+        def get_calc_vals(self, context, data=None):
+            fluid_loop_b = context.BASELINE_0
+            temperature_reset_type = fluid_loop_b["heating_design_and_control"][
```
Could we add ```_b``` at the end of variable names in this method since these variables represent values from baseline?

Jiarongx-Xie replied:
Good point. Addressed!

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+            ]["loop_supply_temperature_at_outdoor_low"]
+            return {
+                "temperature_reset_type": temperature_reset_type,
+                "design_outdoor_high_for_loop_supply_reset_temperature": CalcQ(
```
Great to use the ```CalcQ``` function :).

yunjoonjung-PNNL reviewed (approved):
Great job Jiarong. I left a few minor comments. I'd appreciate it if you could see them. Thanks!

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+
+    def create_data(self, context, data):
+        rmi_b = context.BASELINE_0
+        boiler_loop_ids = find_all("boilers[*].loop", rmi_b)
```
Do we need ```$``` (e.g., ```find_all("$.boilers[*].loop", rmi_b)```)?

weilixu reviewed (commented):
Looks good. Really glad you pulled this off. @Jiarongx-Xie great job.

weilixu commented:
We will also need to reformat this PR,

Jiarongx-Xie commented on rct229/rulesets/ashrae9012019/section21/section21rule8.py:
```diff
@@ -0,0 +1,176 @@
+    def create_data(self, context, data):
+        rmi_b = context.BASELINE_0
+        boiler_loop_ids = find_all("boilers[*].loop", rmi_b)
+        return {"loop_boiler_dict": boiler_loop_ids}
```
Good point! addressed.

Jiarongx-Xie commented:
Thank you @yunjoonjung-PNNL  and @weilixu for your timely review and comments. I have addressed these comments. I kindly request your further review. Thanks!
